util/dvsim/BuildOnlyCfg.py

```diff
@@ -0,0 +1,199 @@
+    def _purge(self):
+        if self.scratch_path is not "":
+            try:
+                log.info("Purging scratch path %s", self.scratch_path)
+                os.system("/bin/rm -rf " + self.scratch_path)
+            except IOError:
+                log.error('Failed to purge scratch directory %s',
+                          self.scratch_path)
+
+    def _create_objects(self):
+        # Create build and run modes objects
+        build_modes = Modes.create_modes(BuildModes,
+                                         getattr(self, "build_modes"))
+        setattr(self, "build_modes", build_modes)
+
+        # All defined build modes are being built, h
+        # ence extend all with the global opts.
+        for build_mode in build_modes:
+            build_mode.build_opts.extend(self.build_opts)
+
+    def _print_list(self):
+        for list_item in self.list_items:
+            log.info("---- List of %s in %s ----", list_item, self.name)
+            if hasattr(self, list_item):
+                items = getattr(self, list_item)
+                for item in items:
+                    log.info(item)
+            else:
+                log.error("Item %s does not exist!", list_item)
+
+    def _create_dirs(self):
+        '''Create initial set of directories
+        '''
+        # Invoking system calls has a performance penalty.
+        # Construct a single command line chained with '&&' to invoke
+        # the system call only once, rather than multiple times.
+        create_link_dirs_cmd = ""
+        for link in self.links.keys():
+            create_link_dirs_cmd += "/bin/rm -rf " + self.links[link] + " && "
+            create_link_dirs_cmd += "mkdir -p " + self.links[link] + " && "
+        create_link_dirs_cmd += " true"
+
+        try:
+            os.system(create_link_dirs_cmd)
+        except IOError:
+            log.error("Error running when running the cmd \"%s\"",
+                      create_link_dirs_cmd)
+            sys.exit(1)
+
+    def _create_deploy_objects(self):
+        '''Create deploy objects from build modes
+        '''
+        builds = []
+        build_map = {}
+        for build in self.build_modes:
+            item = CompileBuildOnly(build, self)
+            builds.append(item)
+            build_map[build] = item
+
+        self.builds = builds
+        self.deploy = builds
+
+        # Create initial set of directories before kicking off the regression.
+        self._create_dirs()
+
+    def create_deploy_objects(self):
+        '''Public facing API for _create_deploy_objects().
+        '''
+        super().create_deploy_objects()
+
+    # deploy additional commands as needed. We do this separated for coverage
+    # since that needs to happen at the end.
+    def deploy_objects(self):
+        '''This is a public facing API, so we use "self.cfgs" instead of self.
+        '''
+        # Invoke the base class method to run the regression.
+        super().deploy_objects()
```

Comment:
Simply remove the last two methods in this class since you don't override anything.

Reply:
removed.